mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/ClientLoanIntegrationTest.java

```diff
@@ -4558,5 +4560,140 @@ private Integer createSavingsProduct(final RequestSpecification requestSpec, fin
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "interestType"));
+    	assertEquals(this.loanTransactionHelper.getLoanProductDetail(this.requestSpec, this.responseSpec, loanProductID, "transactionProcessingStrategyId"),
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "transactionProcessingStrategyId"));
+    	assertEquals(this.loanTransactionHelper.getLoanProductDetail(this.requestSpec, this.responseSpec, loanProductID, "interestCalculationPeriodType"),
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "interestCalculationPeriodType"));
+    	assertEquals(this.loanTransactionHelper.getLoanProductDetail(this.requestSpec, this.responseSpec, loanProductID, "repaymentFrequencyType"),
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "repaymentFrequencyType"));
+    	assertEquals(this.loanTransactionHelper.getLoanProductDetail(this.requestSpec, this.responseSpec, loanProductID, "graceOnPrincipalPayment"),
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "graceOnPrincipalPayment"));
+    	assertEquals(this.loanTransactionHelper.getLoanProductDetail(this.requestSpec, this.responseSpec, loanProductID, "graceOnInterestPayment"),
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "graceOnInterestPayment"));
+    	assertEquals(this.loanTransactionHelper.getLoanProductDetail(this.requestSpec, this.responseSpec, loanProductID, "inArrearsTolerance"),
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "inArrearsTolerance"));
+    	assertEquals(this.loanTransactionHelper.getLoanProductDetail(this.requestSpec, this.responseSpec, loanProductID, "graceOnArrearsAgeing"),
+    			this.loanTransactionHelper.getLoanDetail(this.requestSpec, this.responseSpec, loanID, "graceOnArrearsAgeing"));
+    }
+    
+    private JsonObject createLoanProductConfigurationDetailAsTrue(JsonObject loanProductConfigurationAsTrue) {
+    	loanProductConfigurationAsTrue.addProperty("amortizationType", true);
+    	loanProductConfigurationAsTrue.addProperty("interestType", true);
+    	loanProductConfigurationAsTrue.addProperty("transactionProcessingStrategyId", true);
+    	loanProductConfigurationAsTrue.addProperty("interestCalculationPeriodType", true);
+    	loanProductConfigurationAsTrue.addProperty("inArrearsTolerance", true);
+    	loanProductConfigurationAsTrue.addProperty("repaymentEvery", true);
+    	loanProductConfigurationAsTrue.addProperty("graceOnPrincipalAndInterestPayment", true);
+    	loanProductConfigurationAsTrue.addProperty("graceOnArrearsAgeing", true);
+    	return  loanProductConfigurationAsTrue;
+    }
+    
+    private JsonObject createLoanProductConfigurationDetailAsFalse(JsonObject loanProductConfigurationAsFalse) {
```

Comment:
this and the above method has code duplication, would be better to refactor into a single method